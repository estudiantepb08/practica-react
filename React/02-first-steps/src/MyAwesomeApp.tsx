/** UTILIZACION DE VARIABLES 
 * Es recomendable usar las constantes que nunca cambian fuera del componente
 * para evitar que se redefinan en cada renderizado.
*/

import { CSSProperties } from "react";

const name = "Demetrio";
const lastName = "Sinisterra";
const favoriteGame = ['Elden Ring', 'Smash Bros', 'Metal Gear'];
const isActive = true;
const address = {
    zipCode: 'ABC-123',
    country: 'Colombia',
};
const myStyle: CSSProperties = {
    backgroundColor: '#f0f0f0',
    borderRadius: isActive ? 10 : 20, /** Se pueden utilizar condicionales */
    padding: 10,
}

export function MyAwesomeApp() {

    return (
        <>
            <h1>{name}</h1>
            <h2>{lastName}</h2>
            <p>{favoriteGame.join(', ')}</p>
            <p>{(1 + 1) * 3}</p>
            <p>{isActive ? 'Activo' : 'No activo'}</p>
            <h3 style={myStyle}>
                {JSON.stringify(address)}</h3>
        </>
    )
}