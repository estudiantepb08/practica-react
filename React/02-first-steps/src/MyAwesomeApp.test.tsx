import { describe, test, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { MyAwesomeApp } from './MyAwesomeApp';

describe('MyAwesomeApp', () => {
    test('should render firsName and lastName', () => {

        const { container } = render(<MyAwesomeApp />);
        screen.debug();
        // console.log(container.innerHTML);
        const h1 = container.querySelector('h1');
        expect(h1?.innerHTML).toBe('Demetrio');
        const h3 = container.querySelector('h3');
        expect(h3?.innerHTML).toContain('Colombia');
    });

    test('should render firstName and lastName with screen', () => {

        render(<MyAwesomeApp />);
        screen.debug();
        const h1 = screen.getByRole('heading', { level: 1 });
        console.log(h1.innerHTML);
    });

    test('should match with snapshot', () => {
        const { container } = render(<MyAwesomeApp />);
        expect(container).toMatchSnapshot()
    });
});