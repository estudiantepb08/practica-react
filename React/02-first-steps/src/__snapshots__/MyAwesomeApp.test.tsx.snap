// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match with snapshot 1`] = `
<div>
  <h1>
    Demetrio
  </h1>
  <h2>
    Sinisterra
  </h2>
  <p>
    Elden Ring, Smash Bros, Metal Gear
  </p>
  <p>
    6
  </p>
  <p>
    Activo
  </p>
  <h3
    style="background-color: rgb(240, 240, 240); border-radius: 10px; padding: 10px;"
  >
    {"zipCode":"ABC-123","country":"Colombia"}
  </h3>
</div>
`;
